packages/vscode/src/CodeLensProvider.ts

```diff
@@ -0,0 +1,110 @@
+import * as vscode from 'vscode'
+import * as cp from 'child_process'
+
+/**
+ * CodelensProvider
+ */
+export class CodelensProvider implements vscode.CodeLensProvider {
+  private enabled: boolean
+
+  private generatorRegex: RegExp
+  private _onDidChangeCodeLenses: vscode.EventEmitter<void> = new vscode.EventEmitter<void>()
+  public readonly onDidChangeCodeLenses: vscode.Event<void> = this._onDidChangeCodeLenses.event
+
+  constructor() {
+    this.generatorRegex = /(generator [a-zA-Z]+ {)/g
+    this.enabled = vscode.workspace.getConfiguration('prisma').get('enableCodeLens', true)
+
+    vscode.workspace.onDidChangeConfiguration((_) => {
+      this._onDidChangeCodeLenses.fire()
+    })
+  }
+
+  public provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken): vscode.CodeLens[] {
+    if (!this.enabled) {
+      return []
+    }
+
+    const codelenses = this.getCodeLensGenerateSchema(document, token)
+    return ([] as vscode.CodeLens[]).concat(...codelenses)
+  }
+
+  private getCodeLensGenerateSchema(document: vscode.TextDocument, token: vscode.CancellationToken): vscode.CodeLens[] {
+    const generatorRanges = this.getGeneratorRange(document, token)
+
+    const lenses = generatorRanges.map(
+      (range) =>
+        new vscode.CodeLens(range, {
+          title: 'Generate',
+          command: 'prisma.generate',
+          tooltip: `Codelens to run Prisma generate`,
```

Comment:
```suggestion
          tooltip: `Run "prisma generate"`,
```
From a user perspective, I think it makes more sense